manifests/init.pp

```diff
@@ -1,113 +1,38 @@
-#
-# === Examples
-#
-#  class { 'nscd': }
-#
-# === Authors
-#
-# Steve Traylen <steve.traylen@cern.ch>
-# Arnold Bechtoldt <mail@arnoldbechtoldt.com>
-# Martin Pfeifer <martin.pfeifer@gmail.com>
-#
-# === Copyright
-#
-# Copyright 2011 CERN Steve Traylen
-#
+# nscd class.
+#
+# @summary Configures /etc/ncsd.conf and controls nscd service.
+#
+# @example
+#  include ::nscd
+#
+# @param pkg_ensure state of nscd package.
+# @param service_ensure state of nscd service ensure
+# @param service_enable state of nscd service enable
+# @param nscd_hosts_cache should nscd cache host entries
+# @param nscd_passwd_cache should nscd cache passwd entries
+# @param nscd_group_cache should nscd cache group entries
+# @param nscd_service_cache should nscd cache service entries.
+# @param nscd_threads number of threads
+# @prarm nscd_paranoia  enable internal restart mode.
+# @param nscd_restart_interval nscd internal restart interval
 class nscd (
-  $pkg_ensure            = 'present',
-  $service_ensure        = 'running',
-  $service_enable        = true,
-  $nscd_hosts_cache      = hiera('nscd_hosts_cache','yes'),
-  $nscd_passwd_cache     = hiera('nscd_passwd_cache','no'),
-  $nscd_group_cache      = hiera('nscd_group_cache','no'),
-  $nscd_service_cache    = hiera('nscd_service_cache','yes'),
-  $nscd_threads          = hiera('nscd_threads','4'),
-  $nscd_paranoia         = hiera('nscd_paranoia','no'),
-  $nscd_restart_interval = hiera('nscd_restart_interval','3600')
+  Enum['present','absent','latest'] $pkg_ensure  = 'present',
+  Boolean           $service_ensure        = true,
+  Boolean           $service_enable        = true,
+  Enum['yes','no']  $nscd_hosts_cache      = 'yes',
+  Enum['yes','no']  $nscd_passwd_cache     = 'no',
+  Enum['yes','no']  $nscd_group_cache      = 'no',
+  Enum['yes','no']  $nscd_service_cache    = 'yes',
+  Integer           $nscd_threads          = 4,
+  Enum['yes','no']  $nscd_paranoia         = 'yes',
+  Integer           $nscd_restart_interval = 3600,
 ) {
 
-  if ! ( $pkg_ensure in ['present','absent','latest'] ) {
-    fail('pkg_ensure must be present, absent or latest')
-  }
-  if ! ( $service_ensure in ['running','stopped'] ) {
-    fail('service_ensure must be running or stopped')
-  }
-  if ! ( $nscd_hosts_cache in ['yes','no'] ) {
-    fail('nscd_hosts_cache must be yes or no')
-  }
-  if ! ( $nscd_passwd_cache in ['yes','no'] ) {
-    fail('nscd_hosts_cache must be yes or no')
-  }
-  if ! ( $nscd_group_cache in ['yes','no'] ) {
-    fail('nscd_hosts_cache must be yes or no')
-  }
-  if ! ( $nscd_service_cache in ['yes','no'] ) {
-    fail('nscd_service_cache must be yes or no')
-  }
-  if ! ( $nscd_paranoia in ['yes','no'] ) {
-    fail('nscd_paranoia must be yes or no')
-  }
-
-  include ::nscd::install
-  include ::nscd::config
-  include ::nscd::service
+  contain ::nscd::install
+  contain ::nscd::config
+  contain ::nscd::service
 
-  anchor{"${name}::begin":}
```

Comment:
I think this means there's no longer a dependency on stdlib?
metadata currently has
```
{ "name": "puppetlabs/stdlib", "version_requirement": ">=2.2.1 < 5" }
```
I was going to suggest bumping to a puppet 4 compatible version, but then figured it can probably just go altogether.